module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1520,6 +1520,23 @@ public function getPickupLocations($patron, $holdInfo = null)
             $limitedServicePoints = $this->getLocationData($itemLocationId)['servicePointIds'];
         }
 
+        // If we have $holdInfo, we can limit ourselves to pickup locations that are valid in context. Because the
+        // allowed service point list doesn't include discovery display names, we can't use it directly; we just
+        // have to obtain a list of IDs to use as a filter below.
+        $legalServicePoints = null;
+        if ($holdInfo) {
+            $allowed = $this->getAllowedServicePoints($this->getInstanceByBibId($holdInfo['id'])->id, $patron['id']);
+            if ($allowed !== null) {
+                $legalServicePoints = [];
+                $preferredRequestType = $this->getPreferredRequestType($holdInfo);
+                foreach ($this->getRequestTypeList($preferredRequestType) as $requestType) {
```

Comment:
I agree that this could use clarification, but I'm also not eager to drastically change the configurations for a minor release (though I could be persuaded to). I think the question is whether we just want to improve the comments, or if we also want to deprecate/rename any settings.

If we just wanted to change the comments, maybe the fallback_request_type setting comment could be changed to:

```
; While default_request (and standard driver behavior) determines the preferred request type,
; there are scenarios where the preferred request type is not possible (due to item statuses,
; circulation rules, etc.). This setting lists all request types that the user is allowed to place, in
; priority order from most preferred to least preferred.
```

If we wanted to rename configuration settings, I would perhaps change `fallback_request_type` to `legal_request_types`.

I'm also not sure if we need `default_request` at all, if the order of `fallback_request_type` controls preference... but factoring out that setting would be a more disruptive change.

Reply:
I think the comment improvement is exactly what's needed here, plus maybe a mention in the release notes that people should give that setting a second look.  The configuration setting name is not ideal but I think the comment makes it ok.

> I'm also not sure if we need default_request at all, if the order of fallback_request_type controls preference... but factoring out that setting would be a more disruptive change.

Yeah I agree on both parts of this.  Maybe a change for v11.  The problem is that default_request has a non-empty default in the code (Hold) whereas fallback_request_type defaults to empty.  So if we remove default_request, we really have to change the default (in either config or code) for the other.

Reply:
I've added a TODO checkbox to update the changelog so I won't forget. I'll adjust the comments using the text above next time I have a moment to work on this (at which point I'll also try to get the tests passing again, if you think we're ready to take that step...)

Reply:
Okay, the revised comment has been pushed up. All that's left now is to get tests passing.